src/MessagePack.Annotations/Attributes.cs

```diff
@@ -109,10 +110,13 @@ public UnionAttribute(int key, Type subType)
         /// </summary>
         /// <param name="key">The distinguishing value that identifies a particular subtype.</param>
         /// <param name="subType">The full name (should be assembly qualified) of the derived or implementing type.</param>
+#if NET8_0_OR_GREATER
+        [RequiresUnreferencedCode("The type named in 'subType' may be trimmed.")]
+#endif
         public UnionAttribute(int key, string subType)
         {
             this.Key = key;
-            this.SubType = Type.GetType(subType, throwOnError: true);
+            this.SubType = Type.GetType(subType, throwOnError: true) ?? throw new Exception("SubType not found");
```

Comment:
Perhaps never. But I added it to suppress a warning. Probably because [Type.GetType(string, bool) is documented as returning Type?](https://apisof.net/catalog/22ba44bffb55f7fb26793cf2ae58aa79).